spring-kafka/src/main/java/org/springframework/kafka/support/serializer/JsonDeserializer.java

```diff
@@ -53,15 +53,17 @@
 public class JsonDeserializer<T> implements ExtendedDeserializer<T> {
 
 	/**
+	 * Kafka config property for the default key type if no header.
+	 *
 	 * @deprecated in favor of {@link #KEY_DEFAULT_TYPE}
 	 */
 	@Deprecated
 	public static final String DEFAULT_KEY_TYPE = "spring.json.key.default.type";
 
 	/**
-	 * @deprecated in favor of {@link #VALUE_DEFAULT_TYPE}
+	 * Kafka config property for the default value type if no header.
+	 * <p>Will be removed in the future, use {@link #VALUE_DEFAULT_TYPE} instead.
 	 */
-	@Deprecated
```

Comment:
No, you didn't understand me. 
We definitely have to deprecate it to encourage end-users to migrate to a new value.
My point was to use `"spring.json.default.value.type"` literal instead of this constant in the `else if` below.